tests/integration/targets/mso_backup/tasks/main.yml

```diff
@@ -0,0 +1,295 @@
+    state: present
+      
+- name: Ensure schema 1 with Template 1 exists
+  mso_schema_template: 
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    tenant: ansible_test
+    template: Template 1
+    state: present
+
+- name: Ensure schema 1 with Template 2 exists
+  mso_schema_template: 
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    tenant: ansible_test
+    template: Template 2
+    state: present
+
+- name: Ensure VRF1 exists
+  mso_schema_template_vrf: 
+    <<: *mso_info
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template 1
+    vrf: VRF1
+    state: present
+
+- name: Ensure Template 1 with AP1 exists
+  mso_schema_template_anp:
+    <<: *mso_info 
+    schema: '{{ mso_schema | default("ansible_test") }}'
+    template: Template 1
+    anp: AP1
+    state: present
+
+# Create Backup TO DO MSO Error?
+- name: Create Backup1 in check mode
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup1
+    description: via Ansible
+    location_type: local
+    state: present
+  register: b_cm
+  check_mode: yes
+
+- name: Verify b_cm
+  assert:
+    that:
+    - b_cm is changed
+
+- name: Create Backup1 in normal mode
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup1
+    description: via Ansible
+    location_type: local
+    state: present
+  register: b_nm
+
+- name: Verify b_nm
+  assert:
+    that:
+    - b_nm.current.backupEntry.metadata.name is match("Backup1_[0-9a-zA-Z]*")
+
+- name: Create Backup2 in normal mode
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup2
+    description: via Ansible
+    location_type: local
+    state: present
+  register: b2_nm
+
+- name: Verify b2_nm
+  assert:
+    that:
+    - b2_nm.current.backupEntry.metadata.name is match ("Backup2_[0-9a-zA-Z]*")
+
+- name: Create Backup3 in normal mode
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup3
+    description: via Ansible
+    location_type: local
+    state: present
+  register: b3_nm
+
+- name: Verify b3_nm
+  assert:
+    that:
+    - b3_nm.current.backupEntry.metadata.name is match ("Backup3_[0-9a-zA-Z]*")
+
+- name: Create Backup3 in normal mode again
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup3
+    description: via Ansible
+    location_type: local
+    state: present
+  ignore_errors: yes
+  register: b3_nm_again
+
+- name: Verify b3_nm_again
+  assert:
+    that:
+    - b3_nm_again.current.backupEntry.metadata.name is match ("Backup3_[0-9a-zA-Z]*")
+
+# Query a Backup TO DO
+- name: Query Backup3
+  mso_backup:
+    <<: *mso_info 
+    backup: Backup3
+    state: query
+  register: b3_qm
+
+- name: Verify b3_qm
+  assert:
+    that:
+    - b3_qm is not changed
+    - b3_qm.current | length == 2
+
+# Query All
+- name: Query All
+  mso_backup:
+    <<: *mso_info 
+    state: query
+  ignore_errors: yes
+  register: b_qm_all
+
+- name: Verify b_qm_all
+  assert:
+    that:
+    - b_qm_all is not changed
```

Comment:
You should check the names of the backups you already know or at least that the number is there.